core/src/main/java/bio/terra/pearl/core/service/notification/substitutors/EnrolleeEmailSubstitutor.java

```diff
@@ -173,15 +175,39 @@ public String getParticipantSupportEmailLink(PortalEnvironment portalEnvironment
     /**
      * gets a link the participant can use to create their b2c account, given that they already exist in Juniper
      */
-    public String getInvitationLink(PortalEnvironment portalEnv, PortalEnvironmentConfig config, String portalShortcode, ParticipantUser participantUser) {
+    public String getInvitationLink(PortalEnvironment portalEnv,
+                                    PortalEnvironmentConfig config,
+                                    String portalShortcode,
+                                    ParticipantUser participantUser,
+                                    Profile profile,
+                                    boolean isProxy) {
         try {
-            return "%s%s?accountName=%s".formatted(
+            String username = participantUser != null ? participantUser.getUsername() : "";
+
+            if (isProxy) {
+                username = removeProxySuffix(username);
+            }
+
+            String url = "%s%s?accountName=%s".formatted(
                     routingPaths.getParticipantBaseUrl(portalEnv, config, portalShortcode),
                     routingPaths.getParticipantInvitationPath(),
-                    participantUser != null ?
-                            URLEncoder.encode(participantUser.getUsername(), StandardCharsets.UTF_8.toString()) : "");
+                            URLEncoder.encode(
+                                    username,
+                                    StandardCharsets.UTF_8.toString()));
+
+            if (profile != null && StringUtils.isNotEmpty(profile.getPreferredLanguage())) {
+                url += "&preferredLanguage=" + profile.getPreferredLanguage();
+            }
+            return url;
         } catch (UnsupportedEncodingException e) {
             throw new IOInternalException("unable to encode username");
         }
     }
+
+    private String removeProxySuffix(String email) {
+        // if the email ends with -prox-XXXX, remove it
+        Pattern proxySuffix = Pattern.compile("-prox-[A-Z]{4}$");
```

Comment:
I thought about getting the email from the main account... but it made `EnrolleeContext` more complicated and it's already gotten really big.

Reply:
Let's put this method in RegistrationService so that the correlation between it and `getGovernedUsername` is more overt.  (or consider moving both to a `ProxyUsernameUtils` or somesuch)